### PR TITLE
feat(configuration): Abstract Configuration implementation

### LINKED ISSUE #2: Create Configuration APIs from existing Registry APIs
Populate this repo with the Configuration APIs from go-mod-register Per the Registry Refactoring ADR (https://github.com/edgexfoundry/edgex-docs/blob/master/design/adr/0001-Registy-Refactor.md). Name refactoring will be required.

### PR DESCRIPTION
**Pull request by lenny-intel**
Code was extracted from the abstract Registry implementation in go-mod-registry. Then applied these changes:
- Name refactoring to change from `Registry` to `Configuration`
- Removed any Service Registry specific code/variables/unit tests/mocks
- Addressed https://github.com/edgexfoundry/go-mod-registry/issues/14

closes #2

### CODE REVIEW

**Comment by michaelestrin** [comment]
On leave until January 6th; will review if still open then.

**Comment by cloudxxx8** [comment]
add @hahattan for awareness

**Comment by michaelestrin** [comment]
@lenny-intel 

Approved; presumed this duplicates existing code in go-mod-registry with minor naming changes.   Did not merge as I'm the only one on a list of 6 reviewers that has reviewed the code.  

Note build pipeline does not currently run tests; it only checks for DCO.

image

**Comment by lenny-intel** [comment]
@michaelestrin, correct. The pipeline CI verify stuff will need to be added when we move this out of holding.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module github.com/edgexfoundry/go-mod-configuration
+
+require (
+	github.com/cenkalti/backoff v2.2.1+incompatible // indirect
+	github.com/hashicorp/consul/api v1.1.0
+	github.com/mitchellh/consulstructure v0.0.0-20190329231841-56fdc4d2da54
+	github.com/mitchellh/copystructure v1.0.0 // indirect
+	github.com/pelletier/go-toml v1.2.0
+	github.com/stretchr/testify v1.3.0
+)
```
Comment (tsconn23) [inline go.mod]: Should have minimum Go version declared at the end. Example

https://github.com/edgexfoundry/go-mod-core-contracts/blob/93abd0d268c8b86928b371cca9cd9e94d7b05568/go.mod#L12

**Comment by lenny-intel** [comment]
@tsconn23, minimum Go version added. Can your re-approve and I'll merge.

**Pull request merged**